cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperatorTest.java

```diff
@@ -407,12 +406,15 @@ protected boolean shouldIncrementGeneration(StatefulSetDiff diff) {
             }
         };
 
-        op.deleteAsync("myns", "mysts", true).setHandler(res -> {
+        Async async = context.async();
+        op.deleteAsync(NAMESPACE, RESOURCE_NAME, true).setHandler(res -> {
             if (res.succeeded())    {
                 assertTrue(cascadingCaptor.getValue());
             } else {
                 fail();
```

Comment:
I think `context.fail()` is better when using an `Async`. Normal junit `fail()` will throw, so the `async.complete()` will never be called. So the test will ultimately fail with a timeout, rather than the `AssertionError` arising from the `fail()`.